companyweb_base/models/companyweb_partner_model.py

```diff
@@ -0,0 +1,244 @@
+
+    cweb_show_button = fields.Boolean(compute="_compute_show_button")
+    cweb_show_tab = fields.Boolean(compute="_compute_show_tab")
+
+    def _compute_show_button(self):
+        for cweb in self:
+            if re.match(r"(BE)?0[0-9]{9}", cweb.vat) and cweb.is_company:
+                cweb.cweb_show_button = True
+            else:
+                cweb.cweb_show_button = False
+
+    def _compute_show_tab(self):
+        for cweb in self:
+            if (
+                cweb.cweb_name_enable
+                or cweb.cweb_jur_form_enable
+                or cweb.cweb_address_enable
+                or cweb.cweb_creditLimit_enable
+                or cweb.cweb_startDate_enable
+                or cweb.cweb_endDate_enable
+                or cweb.cweb_score_enable
+                or cweb.cweb_warnings_enable
+                or cweb.cweb_url_enable
+                or cweb.cweb_vat_liable_enable
+                or cweb.cweb_balance_year_enable
+                or cweb.cweb_prefLang_enable
+                or cweb.cweb_companystatus_enable
+            ):
+                cweb.cweb_show_tab = True
+            else:
+                cweb.cweb_show_tab = False
+
+    def create_hash(self, login, password, secret):
+        today = datetime.now().strftime("%Y%m%d")
+        text = (today + login + password + secret).lower()
+        return sha1(text.encode("utf-8")).hexdigest()
+
+    def button_companyweb(self):
+        user_login = self.env.user.cweb_login
+        user_password = self.env.user.cweb_password
+        print("Current Partner TVA :  %s", self.vat)
+        wizard_form = self.env.ref(
+            "companyweb_base.companyweb_credential_wizard", False
+        )
+        print(
+            "Id for the form : companyweb_base.companyweb_credential_wizard :  %s",
+            wizard_form.id,
+        )
+        if not user_login or not user_login:
+            return {
+                "name": "No Credentials",
+                "type": "ir.actions.act_window",
+                "view_type": "form",
+                "view_mode": "form",
+                "res_model": "companyweb_base.credential_wizard",
+                "view_id": wizard_form.id,
+                "target": "new",
+            }
+
+        client = zeep.Client("https://connect.companyweb.be/V1.3/alacarteservice.asmx")
+        r = client.service.GetCompanyByVat(
+            dict(
+                CompanyWebLogin=user_login,
+                CompanyWebPassword=user_password,
+                ServiceIntegrator=SERVICE_INTEGRATOR_ID,
+                LoginHash=self.create_hash(
+                    user_login, user_password, SERVICE_INTEGRATOR_SECRET
+                ),
+                Language="FR",
+                VatNumber=self.vat,
+            )
+        )
+        print("Return status for the cweb API call :  %s", r["StatusCode"])
+        wizard_form = self.env.ref(
+            "companyweb_base.companyweb_credential_wizard", False
```

Comment:
Remove `, False`: if the wizard is absent, something is really wrong and it's better to fail as soon as possible.